src/Symfony/Bundle/TwigBundle/Resources/config/twig.php

```diff
@@ -79,8 +81,24 @@
         ->set('twig.template_iterator', TemplateIterator::class)
             ->args([service('kernel'), abstract_arg('Twig paths'), param('twig.default_path'), abstract_arg('File name pattern')])
 
+        ->set('twig.template_cache.runtime_cache', FilesystemCache::class)
+            ->args([param('kernel.cache_dir').'/twig'])
+
+        ->set('twig.template_cache.readonly_cache', ReadOnlyFilesystemCache::class)
+            ->args([param('kernel.build_dir').'/twig'])
```

Comment:
The whole mechanism should only be activated when the build and cache dir are different. Otherwise, `ReadOnlyFilesystemCache` will point to the same directory as `FilesystemCache`, which is inefficient. Most projects still work this way.
The `TemplateCacheWarmer` is called during the initial build, only when this directories are different: https://github.com/symfony/symfony/blob/3ec30b693fee9a37cdae6718259395eba2e8b25e/src/Symfony/Component/HttpKernel/Kernel.php#L542-L544

Reply:
Indeed ! I adapted the logic to disable the ChainCache when the build and cache dir are the same 👍